utils/estimated_histogram.hh

```diff
@@ -73,13 +73,59 @@ struct estimated_histogram {
 
     int64_t _count = 0;
     int64_t _sample_sum = 0;
+    int64_t _min = 0;
+    int64_t _max = 0;
 
     estimated_histogram(int bucket_count = 90) {
 
         new_offsets(bucket_count);
         buckets.resize(bucket_offsets.size() + 1, 0);
     }
 
+    int64_t translate_bucket_limit(double a) const {
+        return a;
+    }
+
+    int64_t translate_bucket_limit(const duration& a) const {
+        return std::chrono::duration_cast<std::chrono::microseconds>(a).count();
+    }
+
+    /*!
+     * \brief return a subset of the histogram according to given buckets
+     *
+     * While it is useful to reduce the number of buckets, it's not always clear what
+     * is the best bucket list the caller wants.
+     *
+     * This function, gets a vector of buckets, buckets can be double or duration depends
+     * on the histogram usage and return a minimized histogram
+     */
+    template<class T>
+    seastar::metrics::histogram get_histogram(const std::vector<T>& buckets_limit) const {
```

Comment:
Why not pass base, ratio, and count? It looks too generic.

Also, this shouldn't be a template is there are just two possible Ts. You can have a private implenentation for two non-template public funcitons.

Reply:
the incentive for the change is that exponential is good for lower values, but not so good for higher values.

I'll change to non-template implmentation